runtime/subtype_check_test.go

```diff
@@ -0,0 +1,1720 @@
+	Location:                      location,
+	Identifier:                    "R3",
+	Kind:                          common.CompositeKindResource,
+	Members:                       &sema.StringMemberOrderedMap{},
+	ExplicitInterfaceConformances: []*sema.InterfaceType{interfaceType1, interfaceType2},
+}
+
+// checkSubTypeFunctions calls both checkSubTypeWithoutEquality and checkSubTypeWithoutEquality_gen
+// and asserts they produce the same result
+func checkSubTypeFunctions(t *testing.T, subType sema.Type, superType sema.Type) bool {
+	result := sema.CheckSubTypeWithoutEquality(subType, superType)
+
+	generatedSemaResult := sema.CheckSubTypeWithoutEquality_gen(subType, superType)
+	assert.Equal(
+		t,
+		result,
+		generatedSemaResult,
+		"generated function in `sema` package produced different results for"+
+			" subType=%s, superType=%s: manual=%s, generated=%s",
+		subType,
+		superType,
+		result,
+		generatedSemaResult,
+	)
+
+	elaboration := sema.NewElaboration(nil)
+
+	elaboration.SetInterfaceType(interfaceType1.ID(), interfaceType1)
+	elaboration.SetInterfaceType(interfaceType2.ID(), interfaceType2)
+	elaboration.SetInterfaceType(interfaceType3.ID(), interfaceType3)
+	elaboration.SetInterfaceType(interfaceTypeWithConformance1.ID(), interfaceTypeWithConformance1)
+
+	elaboration.SetCompositeType(structType1.ID(), structType1)
+	elaboration.SetCompositeType(structType2.ID(), structType2)
+	elaboration.SetCompositeType(structTypeWithConformance1.ID(), structTypeWithConformance1)
+	elaboration.SetCompositeType(structTypeWithConformance2.ID(), structTypeWithConformance2)
+
+	elaboration.SetCompositeType(resourceType1.ID(), resourceType1)
+	elaboration.SetCompositeType(resourceTypeWithConformance1.ID(), resourceTypeWithConformance1)
+	elaboration.SetCompositeType(resourceTypeWithConformance2.ID(), resourceTypeWithConformance2)
+
+	elaboration.SetEntitlementType(entitlementType.ID(), entitlementType)
+
+	subStaticType := interpreter.ConvertSemaToStaticType(nil, subType)
+
+	superStaticType := interpreter.ConvertSemaToStaticType(nil, superType)
+
+	inter, err := interpreter.NewInterpreter(
+		nil,
+		common.AddressLocation{},
+		&interpreter.Config{
+			CompositeTypeHandler: func(location common.Location, typeID interpreter.TypeID) *sema.CompositeType {
+				return elaboration.CompositeType(typeID)
+			},
+			ImportLocationHandler: func(inter *interpreter.Interpreter, location common.Location) interpreter.Import {
+				return interpreter.VirtualImport{
+					Elaboration: elaboration,
+				}
+			},
+		},
+	)
+	require.NoError(t, err)
+
+	generatedStaticTypeResult := interpreter.CheckSubTypeWithoutEquality_gen(
+		inter,
+		subStaticType,
+		superStaticType,
+	)
+	assert.Equal(
+		t,
+		result,
+		generatedStaticTypeResult,
+		"generated function in `interpreter` package produced different results for"+
+			" subType=%s, superType=%s: manual=%s, generated=%s",
+		subType,
+		superType,
+		result,
+		generatedStaticTypeResult,
+	)
```

Comment:
New addition: Now this test also tests the `interpreter.CheckSubTypeWithoutEquality_gen(...)` function